test_conformance/api/negative_queue.cpp

```diff
@@ -172,3 +350,259 @@ int test_negative_create_command_queue_with_properties_khr(
                            TEST_FAIL);
     return TEST_PASS;
 }
+
+int test_negative_set_default_device_command_queue(cl_device_id deviceID,
+                                                   cl_context context,
+                                                   cl_command_queue queue,
+                                                   int num_elements)
+{
+    cl_int err = 0;
+    if (get_device_cl_version(deviceID) >= Version(3, 0))
+    {
+        cl_device_device_enqueue_capabilities device_capabilities = 0;
+        cl_int err = clGetDeviceInfo(
+            deviceID, CL_DEVICE_DEVICE_ENQUEUE_CAPABILITIES,
+            sizeof(device_capabilities), &device_capabilities, nullptr);
+        test_error(err, "clGetDeviceInfo");
+        if (((device_capabilities & CL_DEVICE_QUEUE_REPLACEABLE_DEFAULT) == 0)
+            && ((device_capabilities & CL_DEVICE_QUEUE_SUPPORTED) == 1))
+        {
+            constexpr int NUM_QUEUE_PROPERTIES = 5;
+            const std::array<cl_queue_properties, NUM_QUEUE_PROPERTIES>
+                properties = { CL_QUEUE_PROPERTIES,
+                               CL_QUEUE_ON_DEVICE_DEFAULT | CL_QUEUE_ON_DEVICE
+                                   | CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE,
+                               0 };
+            clCommandQueueWrapper cmd_queue =
+                clCreateCommandQueueWithProperties(context, deviceID,
+                                                   properties.data(), &err);
+            test_error(err, "clCreateCommandQueueWithProperties");
+            err = clSetDefaultDeviceCommandQueue(context, deviceID, cmd_queue);
+            test_failure_error_ret(
+                err, CL_INVALID_OPERATION,
+                "clSetDefaultDeviceCommandQueue should return "
+                "CL_INVALID_OPERATION when \"device does not support a "
+                "replaceable default on-device queue\"",
+                TEST_FAIL);
+        }
+    }
+
+    err = clSetDefaultDeviceCommandQueue(nullptr, deviceID, queue);
+    test_failure_error_ret(
+        err, CL_INVALID_CONTEXT,
+        "clSetDefaultDeviceCommandQueue should return CL_INVALID_CONTEXT when "
+        "\"context is not a valid context\" using a nullptr",
+        TEST_FAIL);
```

Comment:
I think a lot of these cases need to be updated to allow for `CL_INVALID_OPERATION` as well, when the passed-in device ID does not support a replaceable default on-device queue.  This can happen if the device is a pre-OpenCL 2.0 device, or if the device is an OpenCL 3.0 device that does not support a replaceable default.

Reply:
Done.